src/libtest/lib.rs

```diff
@@ -1549,7 +1804,22 @@ fn calc_result(desc: &TestDesc, task_result: Result<(), Box<dyn Any + Send>>) ->
         (&ShouldPanic::Yes, Ok(())) => TrFailedMsg("test did not panic as expected".to_string()),
         _ if desc.allow_fail => TrAllowedFail,
         _ => TrFailed,
+    };
+
+
+    // If test is already failed (or allowed to fail), do not change the result.
+    if result != TrOk {
```

Comment:
I think you need to check `result != TrOk && result != TrAllowedFail`

Reply:
I'm not sure. If test is allowed to fail, I think user will be confused if that test will anyway be failed in result.

Reply:
I'm not sure about that, either. Probably it doesn't matter because this feature will be removed, #46488.